parser/parser_references.go

```diff
@@ -0,0 +1,116 @@
+package parser
+
+import (
+	"fmt"
+	"go/ast"
+
+	"encr.dev/parser/est"
+	"encr.dev/parser/internal/names"
+	"encr.dev/parser/internal/walker"
+)
+
+func (p *parser) parseResourceUsage() {
+	// For all resources defined, store them in a map per package for faster lookup
+	resourceMap := make(map[string]map[string]est.Resource, len(p.pkgs)) // path -> name -> resource
+	for _, pkg := range p.pkgs {
+		resources := make(map[string]est.Resource, len(pkg.Resources))
+		resourceMap[pkg.ImportPath] = resources
+		for _, res := range pkg.Resources {
+			id := res.Ident()
+			resources[id.Name] = res
+		}
+	}
```

Comment:
Copied from the code which currently tracks `sql.Named` calls - I plan to migrate that to this new path in a future PR (as well as cronjobs)